src/wmtk/TriMeshOperationExecutor.cpp

```diff
@@ -367,6 +367,101 @@ void TriMesh::TriMeshOperationExecutor::replace_incident_face(
 }
 
 Tuple TriMesh::TriMeshOperationExecutor::split_edge()
+{
+    if (!m_mesh.multi_mesh_manager.is_parent_mesh()) {
+        return split_edge_single_mesh();
+    }
+    else
+    {
+        bool is_boundary_on_parent = m_mesh.is_boundary(m_operating_tuple);
+        
+        std::vector<Tuple> vec_t_child;
+        std::vector<Tuple> vec_t_opp_child;
+        vec_t_child = MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, m_operating_tuple);
+        if (!is_boundary_on_parent)
+        {
+            Tuple t_opp = m_mesh.switch_edge(m_operating_tuple);
+            vec_t_opp_child = MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, t_opp);
+        }
+        else
+        {
+            vec_t_opp_child = std::vector<Tuple>(vec_t_child.size(), Tuple()); // fill with null Tuples
+        }
+
+        // do split on parent_mesh
+        Tuple ret_tuple = split_edge_single_mesh();
+
+        for (auto child_mesh_ptr : m_mesh.multi_mesh_manager.child_meshes)
+        {
+            
+            long child_id = child_mesh_ptr->multi_mesh_manager.child_id();
+            Tuple t_child = vec_t_child[child_id];
+            Tuple t_opp_child = vec_t_opp_child[child_id];
+
+            std::vector<long> child_cell_ids_to_update_hash;
+            std::vector<std::pair<long,long>> child_new_cell_ids;
+            if (child_mesh_ptr->top_simplex_type() == PrimitiveType::Face)
+            {
+                // this child_mesh is a TriMesh
+                TriMesh child_tri_mesh = *std::static_pointer_cast<TriMesh>(child_mesh_ptr);
+                if (!t_child.is_null_tuple())
+                {
+                    TriMesh::TriMeshOperationExecutor executor_child(child_tri_mesh, t_child);
+                    executor_child.split_edge();
+                    child_cell_ids_to_update_hash = executor_child.cell_ids_to_update_hash;
+                    
+                    child_new_cell_ids.push_back(std::make_pair(executor_child.m_incident_face_datas[0].split_f1, executor_child.m_incident_face_datas[0].split_f0));
```

Comment:
why is it f1 then f0? seems like an arbitrary convention choice that needs to be discussed